twisted/web/test/test_http2.py

```diff
@@ -1403,6 +1403,32 @@ def validate(streamID):
         return a._streamCleanupCallbacks[1].addCallback(validate)
 
 
+    def test_resetAfterBody(self):
+        """
+        A client that immediately resets after sending the body causes Twisted
+        to send no response.
+        """
+        f = FrameFactory()
+        transport = StringTransport()
+        a = H2Connection()
+        a.requestFactory = DummyHTTPHandler
+
+        requestBytes = f.preamble()
+        requestBytes += buildRequestBytes(self.getRequestHeaders, [], f)
+        requestBytes += hyperframe.frame.RstStreamFrame(
```

Comment:
That API is a different spelling for what is happening here.

Your `h2.makeResetMessage(stream=some_id)` is `hyperframe.frame.RstStreamFrame(stream_id=some_id)`. Your `hyperframe.packForWire(message)` is `frame.serialize()`.